recipes/folly/all/conanfile.py

```diff
@@ -19,216 +19,241 @@ class FollyConan(ConanFile):
+        deps.set_property("zstd", "cmake_file_name", "Zstd")
+        deps.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        folly_deps = os.path.join(self.source_folder, "CMake", "folly-deps.cmake")
+        replace_in_file(self, folly_deps, " MODULE", " ")
+        replace_in_file(self, folly_deps, "OpenSSL 1.1.1", "OpenSSL")
+        # Disable example
+        save(self, os.path.join(self.source_folder, "folly", "logging", "example", "CMakeLists.txt"), "")
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        files.rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
-        files.rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "folly")
         self.cpp_info.set_property("cmake_target_name", "Folly::folly")
         self.cpp_info.set_property("pkg_config_name", "libfolly")
 
-        # TODO: back to global scope in conan v2 once cmake_find_package_* generators removed
-        if Version(self.version) == "2019.10.21.00":
+        if self.settings.os in ["Linux", "FreeBSD"]:
             self.cpp_info.components["libfolly"].libs = [
-                "follybenchmark",
-                "folly_test_util",
-                "folly"
+                "folly_exception_counter",
+                "folly_exception_tracer",
+                "folly_exception_tracer_base",
             ]
-        elif Version(self.version) >= "2020.08.10.00":
-            if self.settings.os == "Linux":
-                self.cpp_info.components["libfolly"].libs = [
-                    "folly_exception_counter",
-                    "folly_exception_tracer",
-                    "folly_exception_tracer_base",
-                    "folly_test_util",
-                    "follybenchmark",
-                    "folly"
-                ]
-            else:
-                self.cpp_info.components["libfolly"].libs = [
-                    "folly_test_util",
-                    "follybenchmark",
-                    "folly"
-                ]
+        self.cpp_info.components["libfolly"].libs += [
+            "folly_test_util",
+            "follybenchmark",
+            "folly"
+        ]
```

Comment:
I don't know why users of libfolly component should link against these other supporing libraries. To me this is a source of issues.
```suggestion
        self.cpp_info.components["libfolly"].libs = ["folly"]
```